pageserver/src/tenant/tasks.rs

```diff
@@ -54,31 +54,21 @@ impl BackgroundLoopKind {
     }
 }
 
-pub(crate) enum RateLimitError {
-    Cancelled,
-}
-
-pub(crate) async fn concurrent_background_tasks_rate_limit(
+pub(crate) async fn concurrent_background_tasks_rate_limit_permit(
```

Comment:
Should have cancellation-safety comment.

Reply:
Feels extra since it's just a semaphore but adding.

Reply:
5885bb1c6